sup3r/qa/feature_plots.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+"""Module to plot feature output from forward passes for visual inspection"""
+import numpy as np
+import matplotlib.pyplot as plt
+import logging
+import glob
+import json
+from datetime import datetime as dt
+
+from rex import MultiFileResourceX
+from rex.utilities.fun_utils import get_fun_call_str
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+from sup3r.utilities import ModuleName
+
+
+logger = logging.getLogger(__name__)
+
+
+class Sup3rVisualQa:
+    """Module to plot features for visual qa"""
+
+    def __init__(self, file_paths, out_pattern, features, time_step=10,
+                 workers=None, **kwargs):
+        """
+        Parameters
+        ----------
+        file_paths : list | str
+            Specifies the files to use for the plotting routine. This is either
+            a list of h5 files generated by the forward pass module or a string
+            pointing to h5 forward pass output which can be parsed by glob.glob
+        out_pattern : str
+            The pattern to use for naming the plot figures. This must include
+            {feature} and {index} so output files can be named with
+            out_pattern.format(feature=feature, index=index). The number of
+            plot figures is determined by the time_index of the h5 files and
+            the time_step argument. The index key refers to the plot file index
+            from the list of all plot files generated.
+        features : list
+            List of features to plot from the h5 files provided.
+        time_step : int
+            Number of timesteps to average over for a single plot figure.
+        workers : int | None
+            Max number of workers to use for plotting. If workers=1 then all
+            plots will be created in serial.
+        **kwargs : dict
+            Dictionary of kwargs passed to matplotlib.pyplot.scatter().
+        """
+
+        self.features = features
+        self.out_pattern = out_pattern
+        self.time_step = time_step
+        self.file_paths = (file_paths if isinstance(file_paths, list)
+                           else glob.glob(file_paths))
+        self.workers = workers
+        self.kwargs = kwargs
+
+    def run(self):
+        """
+        Create plot figures for all the features in self.features. For each
+        feature there will be n_files created, where n_files is the number of
+        timesteps in the h5 files provided divided by self.time_step.
+        """
+        with MultiFileResourceX(self.file_paths) as res:
```

Comment:
I would recommend that the ResourceClassHandler be a kwarg. We almost never use the "*X" resource handlers actually. The "extraction" handlers mostly include extra features for external users who want to query based on lat/lon and whatnot. They are not subclasses of the `MultiFileResource` handler (for example) and do not have as many useful features for our use cases.

Reply:
Ah gotcha, good call.